### PR TITLE
add registryMirrors

### PR DESCRIPTION
**Pull request by qqshfox**
Since the network access to the official registry is quite slow in China, we usually use a mirror instead. But currently there is no way to set this options.

### CODE REVIEW

**Comment by chrislovecnm** [comment]
The PR that I have open for self hosting all the assets will assist with this. You will be able to point to which ever repo you want for either binaries or containers. 
https://github.com/kubernetes/kops/pull/2573

I am currently refactoring it based on another pr that will come in first.

**Comment by qqshfox** [comment]
@chrislovecnm not only will the mirrors benefit the cluster provisioning but also all the images pulling for any pods. I think it's different purpose with the PR you mentioned.

File: pkg/apis/kops/v1alpha1/dockerconfig.go [hunk 1]
```diff
@@ -29,8 +29,9 @@ type DockerConfig struct {
 	// But nodeup will also process a comma-separate list, selecting the first supported option
 	Storage *string `json:"storage,omitempty" flag:"storage-driver"`
 
-	InsecureRegistry *string `json:"insecureRegistry,omitempty" flag:"insecure-registry"`
-	MTU              *int32  `json:"mtu,omitempty" flag:"mtu"`
+	InsecureRegistry *string  `json:"insecureRegistry,omitempty" flag:"insecure-registry"`
+	RegistryMirrors  []string `json:"registryMirrors,omitempty" flag:"registry-mirror,repeat"`
```
Comment (chrislovecnm) [inline pkg/apis/kops/v1alpha1/dockerconfig.go]: Can we get code level documentation for this?

**Comment by qqshfox** [comment]
> * documentation, you can include examples in cluster spec document.
Done.
> * code level docs on Registry Mirrors
Done.
> * I think we have docker unit tests, so please update the test, or file and issue to provide a test later on.
Added a test to `populatecluster_test.go`.

> How have you tested this? Which k8s versions?
Yes. On v1.6.2.

**Comment by justinsb** [comment]
LGTM, and this is the approach used in GCE kube-up (for example): https://github.com/kubernetes/kubernetes/blob/8316bbc14ce822ae8de6707a71396fb58dafe1f1/cluster/gce/configure-vm.sh#L786

**Comment by justinsb** [comment]
Thanks @qqshfox

**Pull request merged**